File: ExternalString/CS/ExternalString.cs
namespace NumbatLogic
{
	class ExternalString
	{
		public static bool Equal(string szA, string szB)
		{
			return string.Equals(szA, szB);
		}

		public static uint GetChecksum(string szString)
		{
			ulong hash = 0;
			for (int i = 0; i < szString.Length; i++)
			{
				ulong c = szString[i];
				hash = c + (hash << 6) + (hash << 16) - hash;
			}
			return (uint)hash;
		}

		public static long hextol(string szString)
		{
			return long.Parse(szString, System.Globalization.NumberStyles.HexNumber);
		}

		public static int atoi(string szString)
		{
			return int.Parse(szString);
		}

		public static long atol(string szString)
		{
			return long.Parse(szString);
		}

		public static double atof(string szString)
		{
			return double.Parse(szString);
		}

	}
}

File: ExternalString/CPP/ExternalString.cpp
#include "ExternalString.hpp"

#include <stdio.h>
#include <stdlib.h>
#include <cstring>

namespace NumbatLogic
{
	bool ExternalString::Equal(const char* szA, const char* szB)
	{
		return strcmp(szA, szB) == 0;
	}

	// a bad checksum
	int ExternalString::GetChecksum(const char* szString)
	{
		int nResult = 0xABC123;
		const char* p = szString;
		while (*p)
			nResult = (nResult ^ *(p++)) << 1;
		return nResult;
	}

	long ExternalString::hextol(const char* szString)
	{
		return ::strtol(szString, NULL, 16);
	}

	int ExternalString::atoi(const char* szString)
	{
		return ::atoi(szString);
	}

	double ExternalString::atof(const char* szString)
	{
		return ::atof(szString);
	}

	long ExternalString::atol(const char* szString)
	{
		return ::atol(szString);
	}
}